packages/babel-plugin-transform-new-target/src/index.ts

```diff
@@ -36,12 +43,11 @@ export default declare(api => {
           }
 
           const { node } = func;
+          if (t.isMethod(node)) {
```

Comment:
A method must not have id so this step can be advanced, which also helps narrowing down the types.